packages/gatsby/src/redux/reducers/pages.ts

```diff
@@ -1,31 +1,44 @@
-const normalize = require(`normalize-path`)
+import normalize from "normalize-path"
+import { ActionsUnion, IGatsbyPage, IGatsbyState } from "../types"
 
-module.exports = (state = new Map(), action) => {
+export const pagesReducer = (
+  state: IGatsbyState["pages"] = new Map<string, IGatsbyPage>(),
+  action: ActionsUnion
+): IGatsbyState["pages"] => {
   switch (action.type) {
     case `DELETE_CACHE`:
       return new Map()
+
     case `CREATE_PAGE`: {
       action.payload.component = normalize(action.payload.component)
-      if (!action.plugin && !action.plugin.name) {
+
+      // throws an error if the page is not created by a plugin
+      if (!action.plugin?.id) {
         console.log(``)
         console.error(JSON.stringify(action, null, 4))
         console.log(``)
+
```

Comment:
```suggestion
      if (!action.plugin?.id) {
```
@pieh pointed out that `console.log` can't be used inside a reducer anymore. This would be a good opportunity to remove this.